transformer/cloudfoundrytransformer.go

```diff
@@ -244,8 +250,38 @@ func (t *CloudFoundry) Transform(newArtifacts []transformertypes.Artifact, alrea
 	return nil, artifactsCreated, nil
 }
 
+// NormalizeForEnvironmentVariableName converts the string to be compatible for environment variable name
+func (t *CloudFoundry) normalizeForEnvironmentVariableName(envName string) string {
+	newName := common.DisallowedDNSCharactersRegex.ReplaceAllLiteralString(strings.ToLower(envName), "_")
+	newName = strings.ToUpper(newName)
+	if newName != envName {
+		logrus.Infof("Changing environment name to %s from %s", envName, newName)
+	}
+	return newName
+}
+
+// normalizeEnvironmentVariables normalizes environment variable names
+func (t *CloudFoundry) normalizeEnvironmentVariables(envList []core.EnvVar, vcapMap map[string][]byte) ([]core.EnvVar, map[string][]byte) {
+	newVcapMap := make(map[string][]byte)
+	for index, env := range envList {
+		newEnvName := t.normalizeForEnvironmentVariableName(env.Name)
+		if _, ok := vcapMap[env.Name]; ok {
+			env.ValueFrom.SecretKeyRef.Key = newEnvName
+		}
+		env.Name = newEnvName
+		envList[index] = env
+	}
+	for key, value := range vcapMap {
+		newKey := t.normalizeForEnvironmentVariableName(key)
+		newVcapMap[newKey] = value
+	}
+	return envList, newVcapMap
+}
```

Comment:
Do we need this function? Why not call normalizeForEnvironmentVariableName directly befoer 301?

Reply:
done